src/core/test/benchmarks/src/main/scala/Benchmarks.scala

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import com.microsoft.ml.spark.FileUtilities.{File, readFile, writeFile}
+import org.apache.hadoop.fs.FileSystem
+
+import scala.collection.mutable.ArrayBuffer
+
+abstract class Benchmarks extends TestBase {
+  val moduleName: String
+  val targetDirectory = new File("target")
+  val thisDirectory = {
+    // intellij runs from a different directory
+    val d = new File("src/test/scala")
+    if (d.isDirectory) d else new File(s"$moduleName/src/test/scala")
+  }
+  assert(thisDirectory.isDirectory, "-- the test should run in the sub-project root level")
+  val historicMetricsFile  = new File(thisDirectory, "benchmarkMetrics.csv")
+  val benchmarkMetricsFile = new File(targetDirectory, s"newMetrics_${System.currentTimeMillis}_.csv")
+
+  val accuracyResults = ArrayBuffer.empty[String]
+  def addAccuracyResult(items: Any*): Unit = {
+    val line = items.map(_.toString).mkString(",")
+    println(s"... $line")
+    accuracyResults += line
+    ()
+  }
+
+  def getFileSystem(): FileSystem = {
+    val config = sc.hadoopConfiguration
+    import org.apache.hadoop.fs.Path
+    val dfs_cwd = new Path(".")
+    dfs_cwd.getFileSystem(config)
+  }
+
+  /** Rounds the given metric to 2 decimals.
+    * @param metric The metric to round.
+    * @return The rounded metric.
+    */
+  def round(metric: Double, decimals: Int): Double = {
+    BigDecimal(metric)
+      .setScale(decimals, BigDecimal.RoundingMode.HALF_UP).toDouble
+  }
+
+  def compareBenchmarkFiles(): Unit = {
+    try writeFile(benchmarkMetricsFile, accuracyResults.mkString("\n") + "\n")
+    catch {
+      case e: java.io.IOException => throw new Exception("Not able to process benchmarks file")
+    }
+    val historicMetrics = readFile(historicMetricsFile, _.getLines.toList)
+    if (historicMetrics.length != accuracyResults.length)
+      throw new Exception(s"Mis-matching number of lines in new benchmarks file: $benchmarkMetricsFile")
+    for (((hist,acc),i) <- (historicMetrics zip accuracyResults).zipWithIndex) {
```

Comment:
Might want to use json parser for this work to eliminate this handmade text file parsing

Reply:
can you discuss this with Eli?  I originally output the data to csv in TrainClassifier, but he preferred this way.

Reply:
I definitely think that JSON is an overkill for simple cases like this...

Reply:
CSVs allow users to manipulate these in more complicated ways down the line. Its simple now but in the future someone might want to build on top of this. Dont depend on any huge parsing libraries, but if you can find a lightweight csv reader and writer (or use spark's!) then that would be much easier to work with and understand

Reply:
can you discuss this more with Eli?  I don't have one specific preference or another.  Originally I wrote it with CSV read/write but Eli changed the code to check raw files output to disk, and I liked his new approach.  I could try to go back to my original changes, it would be an easy change, but I think Eli's approach is actually nicer.